source/state_representation/test/tests/test_jacobian.cpp

```diff
@@ -79,6 +79,45 @@ TEST(JacobianTest, TestMutltiplyWithEigen) {
   EXPECT_TRUE(except_thrown);
 }
 
+TEST(JacobianTest, TestMutltiplyJacobian) {
+  Jacobian jac = Jacobian::Random("robot", 7, "test");
+  Jacobian jac2 = Jacobian::Random("robot", 7, "test");
+  Eigen::MatrixXd res = jac * jac2.transpose();
+  EXPECT_TRUE(res.isApprox(jac.data() * jac2.data().transpose()));
+
+  // check with incorrect dimensions
+  std::cout << "here!!!" << std::endl;
+  std::cout << jac.is_compatible(jac2) << std::endl;
+
+  bool except_thrown = false;
+  try {
```

Comment:
Any reason you didn't use `EXPECT_THROW(jac * jac2, IncompatibleSizeException)` for these 3 test cases?